signup/signupController.js

```diff
@@ -106,19 +110,41 @@ define('signupController', [], function() {
 				"siteId": $scope.sharedProperties.selectedSite
 			};
 
-			SignupService.storeAppointments(data).then((response) => {
-				if (typeof response !== 'undefined' && response && response.success){
-					document.body.scrollTop = document.documentElement.scrollTop = 0;
-					$scope.appointmentId = response.appointmentId;
-					$scope.successMessage = $sce.trustAsHtml(response.message);
+			const firstName = $scope.data.firstName || '';
+			const lastName = $scope.data.lastName || '';
+			SignupService.getExistingClientAppointments(firstName, lastName).then((response) => {
+				if (typeof response !== 'undefined' && response && response.success) {
+					// response.numberExistingAppointments === false when query is empty because none exist
+					if(response.numberExistingAppointments === false || response.numberExistingAppointments['numberExistingAppointments'] === '0') {
+						$scope.hasExistingAppointment = false;
+					} else {
+						$scope.hasExistingAppointment = true;
+					}
 
-					// Send the confirmation email
-					if ($scope.data.email != null && $scope.data.email.length > 0) {
-						$scope.emailConfirmation();
+					// Only allow the client to sign up for an appointment if they don't currently have any uncancelled appointments
+					if(!$scope.hasExistingAppointment) {
```

Comment:
This code might read cleaner if you invert some of these ifs (just have to make sure you `return` in this inversion), i.e.
```javascript
if ($scope.hasExistingAppointment) {
    NotificationUtilities.giveNotice('Failure', "You may not sign up for an appointment if you already have an existing one.", false);
    window.location.replace("../cancel/index.php");
    return;
}

// Normal sign-up logic
```

Reply:
Yeah, this is much better. I changed it and I'll test it out